### DISCUSSION TITLE
AR Quick Look does not display call to action on Chrome

### DISCUSSION THREAD

**Discussion by aviggiano** [google/model-viewer#3636] [Q&A]
### Description

_*Disclaimer*: I believe this should be filled as a _Chrome_ bug and not as a _model-viewer_ bug, but since I don't know the means to talk to them, I'm reporting this here since it also affects model-viewer users._

Adding a [call to action](https://developer.apple.com/documentation/arkit/adding_an_apple_pay_button_or_a_custom_action_in_ar_quick_look) button to AR Quick Look only takes effect on mobile Safari, not on Chrome.

As per the Apple documentation, including URL parameters to the USDZ scheme, such as `callToAction`, should create a button on the AR experience on iOS. The problem is that this only works for Safari.

#### Live Demo

https://user-images.githubusercontent.com/3029017/133463279-2d974ff2-cf62-46a6-be7e-30d2dbfd2cbb.mp4

- https://github.com/aviggiano/ar-call-to-action

### Version

- model-viewer: current

### Browser Affected

- [x] Chrome, version: 93.0.4577.39 (Build oficial) stable (64 bits)
- [ ] Edge
- [ ] Firefox
- [ ] IE
- [ ] Safari

### OS

- [ ] Android
- [x] iOS
- [ ] Linux
- [ ] MacOS
- [ ] Windows

### AR

- [ ] WebXR
- [ ] SceneViewer
- [x] QuickLook

**Comment by elalish** [comment]
That's surprising since we use that same hash API to set scale = fixed and it works fine on iOS Chrome. You should be able to append apple's APIs to your `ios-src` in MV  and have them flow into QuickLook; does that work? Does it work from some of their APIs and not others?

**Comment by aviggiano** [comment]
It doesn't work.
I've just updated my MWE, and the bug report remains the same.

See here the code to repro:

```html
  <model-viewer src="https://modelviewer.dev/shared-assets/models/Astronaut.glb"
    ios-src="https://modelviewer.dev/shared-assets/models/Astronaut.usdz#applePayButtonType=buy&checkoutTitle=Biplane%20Toy&checkoutSubtitle=Rustic%20finish%20with%20rotating%20propeller&price=$15"
    alt="A 3D model of an astronaut" ar ar-modes="webxr scene-viewer quick-look" environment-image="neutral" auto-rotate
    camera-controls></model-viewer>
```

See here the video:


https://user-images.githubusercontent.com/3029017/133505360-2815695c-cef7-469c-a007-8342870d9b60.mp4

**Comment by milesgreen** [comment]
There's a related thread here:
https://bugs.chromium.org/p/chromium/issues/detail?id=1163852#c4

iOS Chrome was originally not respecting the fixed scaling parameter. That was fixed.

But they also stated that QuickLook, in a native app, doesn't provide public APIs for handling all the other parameters, and that Safari is using private API for those. 

That was 6 months ago. Maybe things have improved? Worth logging the issue again?
